### PR TITLE
Artist show page

### PR DESCRIPTION
**Pull request by jhgould**
## Description
Add an artist show page and test the Spotify API call with web mocks.  
Resolves #
## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which maintains existing functionality)
- [x ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
## Notes
Josh and I built out the artist show page with an API call to Spotify for the artist's albums. We tested the api with webmocked api. 
## RSpec Results
```
Finished in 0.91472 seconds (files took 2.52 seconds to load)
18 examples, 0 failures
```

### CODE REVIEW

File: app/views/artists/new.html.erb [hunk 1]
```diff
@@ -4,7 +4,9 @@
 <%= label_tag "Name" %><br>
 <%= text_field_tag :name %><br>
 <%= label_tag "Spotify ID" %><br>
-<%= number_field_tag :spotify_id %><br>
+<%= text_field_tag :spotify_id %><br>
+<%= label_tag "Description" %><br>
+<%= text_field_tag :description %><br>
```
Comment (jhgould) [inline app/views/artists/new.html.erb:9]: Add a description column to the artist's table

File: app/views/artists/show.html.erb [hunk 2]
```diff
@@ -1 +1,40 @@
...
+        <ul>
+          <iframe src=<%="https://open.spotify.com/embed/album/#{params[:new_album]}"%> width="300" height="380" frameborder="0" allowtransparency="true" allow="encrypted-media"></iframe><br>
+          <%=button_to "Top Tracks", artist_path(@artist.id), method: :get%>
+        </ul>   
+
+    <%else%>
+        <iframe src=<%="https://open.spotify.com/embed/artist/#{@artist.spotify_id}"%> width="300" height="380" frameborder="0" allowtransparency="true" allow="encrypted-media"></iframe>
+    <%end%>
+</section>
+
```
Comment (jhgould) [inline app/views/artists/show.html.erb:29]: This section renders the player. If an album is clicked the player will render the album's songs and you get an option to go back to top tracks.

File: spec/features/artists/show_spec.rb [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        stub_request(:get, "https://api.spotify.com/v1/artists/1Apw9xiab11PyZLo6YeUoJ/albums").
+        with(
+          headers: {
+        'Accept'=>'*/*',
+        'Accept-Encoding'=>'gzip;q=1.0,deflate;q=0.6,identity;q=0.3',
+        'Authorization'=>'Bearer',
+        'User-Agent'=>'Faraday v1.0.1'
+          }).
+        to_return(status: 200, body: artist_albums, headers: {})
+
```
Comment (jhgould) [inline spec/features/artists/show_spec.rb:22]: Stubbing the API call so the tests will past.

File: app/controllers/artists_controller.rb [hunk 4]
```diff
@@ -15,15 +21,24 @@ def index
...
   def show
     @artist = Artist.find(params[:id])
+    conn = Faraday.new(url: "https://api.spotify.com") do |faraday|
+      faraday.headers["Authorization"] = "Bearer #{current_user.token}"
+    end
+    response = conn.get("/v1/artists/#{@artist.spotify_id}/albums")
+    
+    @artist_albums = JSON.parse(response.body, symbolize_names: true)
+   
   end
```
Comment (jhgould) [inline app/controllers/artists_controller.rb:32]: Good opportunity to refactor and move the API call into a service.

File: app/controllers/artists_controller.rb [hunk 5]
```diff
@@ -15,15 +21,24 @@ def index
 
   def show
     @artist = Artist.find(params[:id])
+    conn = Faraday.new(url: "https://api.spotify.com") do |faraday|
+      faraday.headers["Authorization"] = "Bearer #{current_user.token}"
+    end
+    response = conn.get("/v1/artists/#{@artist.spotify_id}/albums")
+    
+    @artist_albums = JSON.parse(response.body, symbolize_names: true)
+
```
Comment (jhgould) [inline app/controllers/artists_controller.rb:30]: Good opportunity for refactoring. Maybe add it into a service.

File: app/views/artists/show.html.erb (hunk 2 above)
Comment (cgaddis36) [reply app/views/artists/show.html.erb:29]: nice good work here, those params can get tricky

File: spec/features/artists/show_spec.rb (hunk 3 above)
Comment (cgaddis36) [reply spec/features/artists/show_spec.rb:22]: good work here

**Pull request merged**